### ISSUE TITLE
Terraform crash due to invalid JSON

### ISSUE THREAD

**Issue by scjody** [hashicorp/terraform#22704]
### Terraform Version

```
Terraform v0.12.8
```

### Terraform Configuration Files

```hcl
{
  "module": {
    "jody-mp-1": {
      "source": "./main/",
      "providers": {
        "google.cloud": "google.cloud", #TODO
        "google.main": "google.testing"
      },
      "name": "jody-mp-1",
      "type": "testing",
      "license": "nope",
      "password": "also nope",
    }
  }
}
```

### Debug Output

https://gist.github.com/scjody/65c1008cf988faa459b0a20a1d53e240

### Crash Output

https://gist.github.com/scjody/ffadecab541e4e68f566497b7bc62b2d

### Expected Behavior

Some sort of error message - the JSON I'm trying to use is invalid.

### Actual Behavior

Crash

### Steps to Reproduce

1. `terraform init`

**Comment by ashupathshala** [comment]
Hi, did you guys have any update or solution on this

**Comment by alisdair** [comment]
This is an HCL bug which [was fixed upstream](https://github.com/hashicorp/hcl/pull/358) and merged into Terraform in #24650. The fix will be released as part of the 0.13 release. Thanks for reporting!
